### ISSUE TITLE
training data corresponding to the best model in xgbtree

### ISSUE THREAD

**Issue by srinivasan3** [topepo/caret#655]
I am trying to get the training data from best model obtained through caret (method = "xgbTree")

Below is how I construct the model:

set.seed(104)
  
xgb.grid <- expand.grid(nrounds = 1000,
                          eta = c(0.01,0.05,0.1),
                          max_depth = c(2,4,6,8,10,14),gamma = 0, 
                          colsample_bytree = 1, 
                          min_child_weight = 1,subsample=1)
  
ctrl <- trainControl(method = "repeatedcv", repeats = 4,number=5,
                       classProbs = TRUE,
                       summaryFunction = twoClassSummary,
                       sampling = "down",
                       savePredictions=TRUE,
                       verboseIter=FALSE,allowParallel = TRUE)
  
  
  levels <- unique(y)
  targ <- factor(y, labels=make.names(levels))
  
  objModel     <- train(x,targ,  
                        method="xgbTree",metric = "ROC",
                        trControl=ctrl,tuneGrid = xgb.grid,nthread =3)
  
bestmodel = objModel$finalModel

I am able to get the data (down sampled data and its corresponding labels) with other methods such as gbm but not sure how to extract it from xgbTree

**Comment by topepo** [comment]
I don't think that the model object keeps the data. See `?xgboost::xgb.train` at the "Value" section.  

At the worst, you can write a [custom method](https://topepo.github.io/caret/using-your-own-model-in-train.html) and attach `x` and `y` to the model object for use later. You might consider the `final` argument to the fit function too.

**Comment by srinivasan3** [comment]
Thank you Max !
